contracts/src/interfaces/IPayer.sol

```diff
@@ -0,0 +1,598 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title IPayer
+ * @notice Interface for managing payer USDC deposits, usage settlements,
+ *         and a secure withdrawal process with optimized storage using
+ *         Merkle trees and EIP-1283 gas optimizations.
+ */
+interface IPayer {
+    //==============================================================
+    //                             STRUCTS
+    //==============================================================
+
+    /**
+     * @dev Struct to store payer information.
+     * @param balance The current USDC balance of the payer.
+     * @param isActive Indicates whether the payer is active.
+     * @param creationTimestamp The timestamp when the payer was first registered.
+     * @param latestDepositTimestamp The timestamp of the most recent deposit.
+     * @param debtAmount The amount of fees owed but not yet settled.
+     */
+    struct Payer {
+        uint256 balance;
+        bool isActive;
+        uint256 creationTimestamp;
+        uint256 latestDepositTimestamp;
+        uint256 debtAmount;
+    }
+
+    /**
+     * @dev Struct to store withdrawal request information.
+     * @param requestTimestamp The timestamp when the withdrawal was requested.
+     * @param withdrawableTimestamp The timestamp when the withdrawal can be finalized.
+     * @param amount The amount requested for withdrawal.
+     */
+    struct WithdrawalRequest {
+        uint256 requestTimestamp;
+        uint256 withdrawableTimestamp;
+        uint256 amount;
+    }
+
+    /**
+     * @dev Struct to store usage reporting data.
+     * @notice This struct is used primarily for event emission and off-chain tracking.
+     * @dev For storage efficiency, complete reports are NOT stored on-chain; only their
+     *      cryptographic commitments are stored using Merkle trees.
+     * @param payer The address of the payer being charged.
+     * @param nodeId The ID of the node that provided service.
+     * @param messageCount Number of messages processed.
+     * @param dataSize Total size of data processed in bytes.
+     * @param fees The amount charged for this usage.
+     * @param timestamp When the usage occurred.
+     */
+    struct UsageReport {
+        address payer;
+        uint256 nodeId;
+        uint256 messageCount;
+        uint256 dataSize;
+        uint256 fees;
+        uint256 timestamp;
+    }
+
+    /**
+     * @dev Struct to store batch commitment information.
```

Comment:
Can you add a little more context around how batch commitments work

Reply:
So this is an idea aiming to make the protocol auditable from end to end. Completely optional, as we rely on trusted NOPs and the PayerReport attestation process.

We'd submit a PayerReport to the Treasury contract. After being accepted, Treasury calls payer.settleUsageBatch([]payers, []fees) which in turn creates a BatchCommitment storing hash(payers, fees, timestamp), subtract all the funds, etc.
The hash could be used to verify the settlement was processed correctly and noone tampered with the amounts/payers.

Reply:
I see. So it's basically a "receipt" that the settlement was done.